File: VinitKumar/Problems/Milestone2/Problem3.cs
namespace VinitKumar.Problems.Milestone2
{
    public class Problem3 {
        public string MergeAlternately(string word1, string word2) {
            string ans = "";
            if(word1.Length >= word2.Length) {
                for(int i = 0;i < word1.Length;i++) {
                    ans = ans + word1[i];
                    if(i < word2.Length) {
                        ans = ans + word2[i];
                    }
                }
            } else {
                for(int i = 0;i < word2.Length;i++) {
                    if(i < word1.Length) {
                        ans = ans + word1[i];
                    }
                    ans = ans + word2[i];
                }
            }
            return ans;
        }
    }
}

File: SBiswas/Problems/Milestone2/Problem3.cpp
class Solution {
public:
    string mergeAlternately(string word1, string word2) {

        int word1_size = word1.length();
        int word2_size = word2.length();

        int i=0, j=0;
        bool flag=true;
        string ans="";

        while(i<word1_size || j<word2_size)
        {
            if(i>=word1_size)
            {
                // word1 has finised
                ans = ans + word2.substr(j,word2_size-j);
                break;
            }

            if(j>=word2_size)
            {
                // word1 has finised
                ans = ans + word1.substr(i,word1_size-i);
                break;
            }
            
            if(flag)
            {
                ans+=word1[i];
                i++;
            }
            else
            {
                ans+=word2[j];
                j++;
            }

            flag = flag^true;


        }

        return ans;
        
    }
};